src/fu-engine.c

```diff
@@ -1079,11 +1079,29 @@ static gboolean
 fu_engine_check_requirement_firmware (FuEngine *self, XbNode *req,
 				      FuDevice *device, GError **error)
 {
+	FuDevice *device_actual = device;
+	guint64 depth;
 	g_autoptr(GError) error_local = NULL;
 
+	/* look at the parent device */
+	depth = xb_node_get_attr_as_uint (req, "depth");
+	if (depth != G_MAXUINT64) {
```

Comment:
Is this the default return if unset or the extreme? If it's the extreme I think this should be an error condition that it's this value.

Reply:
It's the value for unset.